config/webpack/CustomVersionFilePlugin.ts

```diff
@@ -0,0 +1,26 @@
+import fs from 'fs';
+import path from 'path';
+import type {Compiler} from 'webpack';
+import {version as APP_VERSION} from '../../package.json';
+
+/**
+ * Simple webpack plugin that writes the app version (from package.json) and the webpack hash to './version.json'
+ */
+class CustomVersionFilePlugin {
+    apply(compiler: Compiler) {
+        compiler.hooks.done.tap(this.constructor.name, () => {
+            const versionPath = path.join(__dirname, '/../../dist/version.json');
+            fs.mkdir(path.dirname(versionPath), {recursive: true}, (dirErr) => {
+                if (dirErr) {
+                    console.error('Error creating version.json directory', dirErr);
+                    return;
+                }
+                fs.writeFile(versionPath, JSON.stringify({version: APP_VERSION}), {encoding: 'utf8'}, (err) => {
+                    console.error('Error writing to version.json', err);
```

Comment:
```suggestion
                    if (!err) {
                        return;
                    }
                    throw err;
```
Missing early return